src/litserve/server.py

```diff
@@ -269,6 +305,38 @@ def encode_response(self, outputs):
              """
             )
 
+        if (
+            stream
+            and max_batch_size > 1
+            and not all([
+                inspect.isgeneratorfunction(lit_api.predict),
+                inspect.isgeneratorfunction(lit_api.encode_response),
+                inspect.isgeneratorfunction(lit_api.unbatch),
```

Comment:
we need to add a default implementation for `unbatch` in case of streaming

let's set the streaming flag to the API instance through the server, so we can `return` or `yield` accordingly in `unbatch`

Reply:
@lantiga if I understand correctly we can either `return` or `yield` otherwise the `inspect.isgenerator` won't be able to identify the correctly.

Reply:
right, we probably need to assign the function to the API as LitServe starts